pymongo/common.py

```diff
@@ -611,17 +684,41 @@ def validate(option, value):
     return lower, value
 
 
-def get_validated_options(options, warn=True):
-    """Validate each entry in options and raise a warning if it is not valid.
-    Returns a copy of options with invalid entries removed
+def get_validated_options(options):
+    """Validates and normalizes an options dictionary defined by the user as
+    a list of keyword arguments.
+
+    Returns a new dictionary of validated and normalized options. Errors will
+    be thrown for invalid options.
+
+    :Parameters:
+        - `opts`: A dict of MongoDB options.
+    """
+    return dict(validate(k, v) for k, v in iteritems(options))
+
+
+def get_validated_uri_options(options, warn=True):
+    """Validates and normalizes an options dictionary constructed by parsing
+    a MongoDB URI.
+
+    Returns a new dictionary of validated and normalized options. If warn is
+    False then errors will be thrown for invalid options, otherwise they will
+    be ignored and a warning will be issued.
+
+    :Parameters:
+        - `opts`: A dict of MongoDB URI options.
+        - `warn` (optional): If ``True`` then warnigns will be logged and
+          invalid options will be ignored. Otherwise invalid options will
+          cause errors.
     """
     validated_options = {}
     for opt, value in iteritems(options):
         lower = opt.lower()
         try:
-            validator = URI_VALIDATORS.get(lower, raise_config_error)
+            validator = URI_OPTIONS_VALIDATOR_MAP.get(
+                lower, raise_config_error)
             value = validator(opt, value)
-        except (ValueError, ConfigurationError) as exc:
```

Comment:
Can you extract this change into a separate review for PYTHON-1686?

Reply:
I thought there would be merge conflicts if I resolve these issues separately since this PR changes the names of many of the options affected which is why I didn't bother with a separate PR for 1686. Do we intend to backport these changes? That might make a good case for doing things as you suggest. If not, I'd be in favor of the resolving 1686 after we merge this (shouldn't involve much more than some additional testing) since no users had complained about that anyway.